src/utils/itemValidator.js

```diff
@@ -0,0 +1,27 @@
+export const isInputEmpty = (userInput) => {
+	const trimmedUserInput = userInput.trim();
+	return trimmedUserInput ? false : true;
+};
+
+export const isItemDuplicate = (newItem, itemsList) => {
+	const removePunctuation = (inputItem) => {
```

Comment:
BUG - I think we should consider moving the removePunctuation function out of the scope of the isItemDuplicate function so that it can be used as a validator on its own here and when adding an item for the first time. This is because I found a bug that when I am adding an item that **is not already** on the list I am able to add items with punctuation(even the ones checked for in the regex in this function, as the current logic only looks for if its a duplicate with the punctuation, i believe) and I can also add new items that are not already in the list that have spaces. 

![Screenshot 2024-03-05 at 11 15 09 AM](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/135680082/cc12789d-4643-48e1-b0dc-2301eeaaed9e)

We could also make a ticket later down the line to fix these bugs. I am clearly able to add some undesirable items to the list. For instance, I was able to add "milk&' AND "MiL!k" so if the item on the list has an untracked character in it, such as 'milk&' then it seems the validators aren't able to discern that that is the same as "milk".

Reply:
Hey Paige, 
We though about this and wondered what if someone entered "Half & half". We wouldn't want to prevent the user from adding an item that uses valid punctuation. Or remove the "&" with white space, which would result in "Half half" being added to the list.

For that reason perhaps we leave the punctuation validation out when a user is entering an item thats not already in the list. If they add something like "milk&" and realized after they submitted that they make a typo etc, then a delete item would be the best way to handle this. 

I see there is a deleteItem() function in our firebase config file, and assuming we will implement this soon.

So right now, a user can add "MiL!k" but when they go to add "milk" the second time it won't allow it. Perhaps we can assumption that these types of entries are typos or a user error, and they intended to enter Milk/milk and they'd want to delete "MiL!k".

But, we can still move removePunctionation() outside of isDuplicate(), so that this function can be used independently.

Reply:
Hey Paige, 

I moved removePunctuation() outside of isItemDuplicate().

I also added another validator to check for "and" and "&" - which this discussion made me think would be useful.

This way if a user has "half and half" already in their list and then go to add "half & half" they will get a duplicate alert.

This message doesn't really relate to what your original comment was about, but just thought I'd communicate this new validator with you.